hack/dind-cluster.sh

```diff
@@ -263,12 +267,161 @@ function build-images() {
 function build-image() {
   local build_root=$1
   local image_name=$2
+  local dockerfile=${3:-}
 
+  if [[ -n "${dockerfile}" ]]; then
+    dockerfile="-f ${dockerfile}"
+  fi
   pushd "${build_root}" > /dev/null
-    ${DOCKER_CMD} build -t "${image_name}" .
+    ${DOCKER_CMD} build -t "${image_name}" ${dockerfile} .
   popd > /dev/null
 }
 
+function start-ansible-cluster() {
+  local cluster_id=$1
+  local private_key=$2
+  local force_config_write=$3
+
+  echo "Starting ansible dind cluster '${cluster_id}'"
+  ssh-keygen -N '' -q -f "${private_key}" > /dev/null
+
+  ensure-host-configuration "openshift/dind-host"
+
+  local run_cmd="${DOCKER_CMD} run -d --privileged"
+
+  # Create containers
+  local master_cid; master_cid="$(${run_cmd} --name="${MASTER_NAME}" --hostname="${MASTER_NAME}" "${HOST_IMAGE}")"
+  local master_ip; master_ip="$(get-docker-ip "${master_cid}")"
+  set-authorized-keys "${MASTER_NAME}" "${private_key}"
+  local node_ips=()
+  local cid
+  for name in "${NODE_NAMES[@]}"; do
+    cid="$(${run_cmd} --name="${name}" --hostname="${name}" "${HOST_IMAGE}")"
+    node_ips+=( "$(get-docker-ip "${cid}")" )
+    set-authorized-keys "${name}" "${private_key}"
+  done
+
+  # TODO support a variable number of nodes
+
+  local inventory_file="/etc/ansible/hosts"
```

Comment:
This makes things less reusable. If we are operating under the generic `hack/lib/init.sh` system we have a temp scratchpad at `${BASETMPDIR}`. Ansible can handle an inventory file anywhere.